src/Sparrow/Json/BlittableJsonReaderObject.cs

```diff
@@ -1099,11 +1099,47 @@ protected bool Equals(BlittableJsonReaderObject other)
         {
             if (_propCount != other._propCount)
                 return false;
+            
+            if (_isRoot == false && other._isRoot == false)
+            {
+                var buffer = new PropertiesInsertionBuffer();
+
+                GetPropertiesByInsertionOrder(buffer);
+
+                var dataStart = _objStart - buffer.Offsets[0];
+                var dataSize = _objStart - dataStart;
+
+                other.GetPropertiesByInsertionOrder(buffer);
+
+                var otherDataStart = other._objStart - buffer.Offsets[0];
+                var otherDataSize = other._objStart - otherDataStart;
+
+                if (dataSize != otherDataSize)
+                    return false;
+
+                if (dataSize < int.MaxValue)
+                {
+                    if (Memory.CompareInline(dataStart, otherDataStart, (int)dataSize) == 0)
+                    {
+                        // data of property values is the same, we also need to check names of properties
+                        // they as are defined in root so it requires separate check
+
+                        return HasSamePropertyNames(other);
+                    }
+                }
+            }
+            else if (_isRoot && other._isRoot)
+            {
+                if (_size != other._size)
```

Comment:
No sure if this is always correct assumption. Depending what we decide in http://issues.hibernatingrhinos.com/issue/RavenDB-12122

Reply:
According to tests added there we might have the same blittables but with different sizes. I think the issue is only when creating blittables with the usage of `ManualBlittableJsonDocumentBuilder`

Reply:
We got some test failures because of that: http://jenkins.hibernatingrhinos.com/job/PR_Tests_v4.1/1830/

I dropped this condition in https://github.com/ravendb/ravendb/pull/7805/commits/e5c6a27eda19b1a3b0b15378538c6947f4ca646d